crowbar_framework/app/models/node.rb

```diff
@@ -850,6 +850,22 @@ def file_exist?(file)
     out[:exit_code].zero?
   end
 
+  def upgraded?
+    upgrade_state = @node["crowbar_wall"]["node_upgrade_state"] || ""
+    return false unless upgrade_state == "upgraded"
+    Rails.logger.info("Node #{@node.name} was already upgraded.")
+    true
+  end
+
+  def upgrading?
+    @node["crowbar_wall"]["node_upgrade_state"] == "upgrading"
+  end
+
+  def upgrade_state=(state)
```

Comment:
It is, look at other (old) methods in `node.rb`.
It was actually Hound which forced me to do it this way, I had `set_upgrade_state` method before.